lib/api/initiateMultipartUpload.js

```diff
@@ -144,31 +146,32 @@ function initiateMultipartUpload(authInfo, request, log, callback) {
                 services.metadataStoreMPObject(MPUbucket.getName(),
                     cipherBundle, metadataStoreParams, log, err => {
                         if (err) {
-                            return callback(err);
+                            return callback(err, null, corsHeaders);
                         }
                         pushMetric('initiateMultipartUpload', log, {
                             authInfo,
                             bucket: bucketName,
                         });
-                        return callback(err, xml);
+                        return callback(err, xml, corsHeaders);
                     });
             });
     }
 
     services.metadataValidateAuthorization(metadataValParams,
         (err, destinationBucket) => {
+            corsHeaders = collectCorsHeaders(request, destinationBucket);
```

Comment:
main reason I didn't is because I didn't want to modify the helper function `_storetheMPObject` to add the `corsHeaders` as another param, although looking at it now I definitely could and might as well. I'll update when I've pushed the change.